fastlane_core/spec/ipa_upload_package_builder_spec.rb

```diff
@@ -0,0 +1,26 @@
+describe FastlaneCore do
+  describe FastlaneCore::IpaUploadPackageBuilder do
+    let(:ipa) { 'iOSAppOnly' }
+    let(:path) { File.expand_path("../fixtures/ipas/#{ipa}.ipa", __FILE__) }
+    let(:uploader) { FastlaneCore::IpaUploadPackageBuilder.new }
+    let(:unique_path) { uploader.unique_ipa_path(path) }
+
+    context 'unique IPA file name' do
+      it 'does not contain any special characters' do
+        special_chars = %w[! @ # $ % ^ & * ( ) + = [ ] " ' ; : < > ? / \ | { } , ~ `]
```

Comment:
Ha, nice 👍 Wouldn't it be easier to do it the other way around in this case and use a regex like `[\w\-\.\_]+`?

Reply:
Just thinking out loud here, this test would be green if the file name contained `‘` or `ø`, or an emoji

Reply:
Ah good call @KrauseFx, yeah I think a regex would be the better option!

Reply:
Let's add a test for the test to see if it's covering emojis 😂